### PR TITLE
Scoring validation fixes

### LINKED ISSUE #463: Saving points crashes when DecimalField validation fails
Because the value is not in `cleaned_data` at all

Sentry Issue: [APLIKACJAWWW-2M](https://sentry.io/organizations/warsztatywww/issues/2344649850/?referrer=github_integration)

```
KeyError: 'qualification_result'
(4 additional frame(s) were not displayed)
...
  File "django/forms/forms.py", line 177, in is_valid
    return self.is_bound and not self.errors
  File "django/forms/forms.py", line 172, in errors
    self.full_clean()
  File "django/forms/forms.py", line 375, in full_clean
    self._clean_form()
  File "django/forms/forms.py", line 402, in _clean_form
    cleaned_data = self.clean()
  File "wwwapp/forms.py", line 427, in clean
    if self.cleaned_data['qualification_result'] is not None:
```

### LINKED ISSUE #473: Increase decimal points for grades to 2
Some people like to give grades with 1/4th of a points this results in values like `17.25`. This should be allowed

### PR DESCRIPTION
**Pull request by krzys-h**
* Fix internal server error when score does not fit in DecimalField (closes #463) [semi-critical bugfix]
* Ensure that max_points cannot be negative (reported by @MrQubo a while ago but never made an issue for it)
* Increase decimal points for grades to 2 (closes #473)
* Add some tests to prevent future problems

---
This change is [Reviewable](https://reviewable.io/reviews/warsztatywww/aplikacjawww/476)

### CODE REVIEW

**Review by MrQubo — CHANGES_REQUESTED** [review]
New tests only test `qualification_result` field. `max_points` and `qualification_threshold` aren't tested, and they also belong to a different form (`reverse('workshop_edit')`).

File: wwwapp/forms.py [hunk 1]
```diff
@@ -424,16 +424,13 @@ def clean(self):
         if not self.instance.workshop.is_qualification_editable():
             raise ValidationError('Nie można edytować warsztatów z poprzednich lat')
 
-        if self.cleaned_data['qualification_result'] is not None:
+        if 'qualification_result' in self.cleaned_data and self.cleaned_data['qualification_result'] is not None:
```
Comment (MrQubo) [inline wwwapp/forms.py:427]:
```suggestion
        if self.cleaned_data.get('qualification_result') is not None:
```

**Review by cytadela8 — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 3 of 3 files at r3, 2 of 2 files at r4, 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/warsztatywww/aplikacjawww/476)* status: all files reviewed, 1 unresolved discussion (waiting on @krzys-h)

___
*[wwwapp/tests/test_workshop_qualification.py, line 516 at r2](https://reviewable.io/reviews/warsztatywww/aplikacjawww/476#-MaAR4qFB0wM54H5EmM5:-MaAR4qFB0wM54H5EmM6:b-j1pn80) ([raw file](https://github.com/warsztatywww/aplikacjawww/blob/f5a00bd04fc362e867fcb50eae9e402d90db84a6/wwwapp/tests/test_workshop_qualification.py#L516)):*
> ```Python
>             'qualification_result': 5
>         })
>         self.assertJSONEqual(response.content, {'error': '* qualification_result\n  * Przed wpisaniem wyników, ustaw maksymalną liczbę punktów możliwą do uzyskania'})
> ```

This causes an error? I guess that's fine, but it contributed to the problem.

**Review by cytadela8 — COMMENTED** [review]
:duck: 

> *[Reviewable](https://reviewable.io/reviews/warsztatywww/aplikacjawww/476#-:-MaAV9j7Ai_OI7Nk7K1u:bcmjph0)* status: all files reviewed, 1 unresolved discussion (waiting on @krzys-h)

**Review by MrQubo — APPROVED** [review]

**Review by krzys-h — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/warsztatywww/aplikacjawww/476)* status: all files reviewed, 1 unresolved discussion (waiting on @cytadela8)

___
*[wwwapp/tests/test_workshop_qualification.py, line 516 at r2](https://reviewable.io/reviews/warsztatywww/aplikacjawww/476#-MaAR4qFB0wM54H5EmM5:-MaAXp-X2cPU3dda0Brb:b-lgls2m) ([raw file](https://github.com/warsztatywww/aplikacjawww/blob/f5a00bd04fc362e867fcb50eae9e402d90db84a6/wwwapp/tests/test_workshop_qualification.py#L516)):*
<details><summary><i>Previously, cytadela8 (Artur Puzio) wrote…</i></summary><blockquote>

This causes an error? I guess that's fine, but it contributed to the problem.
</blockquote></details>

Actually it didn't -

**Review by krzys-h — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/warsztatywww/aplikacjawww/476)* status: all files reviewed, 1 unresolved discussion (waiting on @cytadela8)

___
*[wwwapp/tests/test_workshop_qualification.py, line 516 at r2](https://reviewable.io/reviews/warsztatywww/aplikacjawww/476#-MaAR4qFB0wM54H5EmM5:-MaA_xMI3hbV1ZEetBhk:b-p2nmfs) ([raw file](https://github.com/warsztatywww/aplikacjawww/blob/f5a00bd04fc362e867fcb50eae9e402d90db84a6/wwwapp/tests/test_workshop_qualification.py#L516)):*
<details><summary><i>Previously, krzys-h (krzys_h) wrote…</i></summary><blockquote>

Actually it didn't - 
</blockquote></details>

Actually it didn't - if you get to this check, you are already past the point that triggered the bug

**Pull request merged**